tests/phpunit/integration/Modules/Analytics_4/Synchronize_PropertyTest.php

```diff
@@ -0,0 +1,225 @@
+		$this->authentication->get_oauth_client()->set_token( array( 'access_token' => 'valid-auth-token' ) );
+		$this->authentication->verification()->set( true );
+
+		$encrypted_options->set(
+			Credentials::OPTION,
+			array(
+				'oauth2_client_id'     => 'test-client-id',
+				'oauth2_client_secret' => 'test-client-secret',
+			)
+		);
+		add_filter( 'googlesitekit_setup_complete', '__return_true', 100 );
+
+		$permissions->register();
+
+		$this->analytics_4 = new Analytics_4(
+			$context,
+			$this->options,
+			$this->user_options,
+			$this->authentication
+		);
+
+		$this->set_analytics_settings();
+
+		$this->synchronize_property = new Synchronize_Property(
+			$this->analytics_4,
+			$this->user_options,
+			$this->authentication
+		);
+	}
+
+	public function set_analytics_settings() {
+		// Ensure Anlytics 4 module is connected
+		$this->options->set(
+			Analytics_4_Settings::OPTION,
+			array(
+				'accountID'       => '12345678',
+				'propertyID'      => '987654321',
+				'webDataStreamID' => '1234567890',
+				'measurementID'   => 'G-A1B2C3D4E5',
+			)
+		);
+
+		// Set owner ID, using base analytics settings.
+		$this->options->set(
+			Settings::OPTION,
+			array(
+				'ownerID' => get_current_user_id(),
+			)
+		);
+	}
+
+	public function fake_analytics_4_property_object( $property_id, $create_time ) {
+		FakeHttp::fake_google_http_handler(
+			$this->analytics_4->get_client(),
+			function ( Request $request ) use ( $property_id, $create_time ) {
+				$url = parse_url( $request->getUri() );
+
+				$property = new GoogleAnalyticsAdminV1betaProperty();
+				$property->setCreateTime( $create_time );
+				$property->setName( "properties/{$property_id}" );
+
+				if ( "/v1beta/properties/{$property_id}" === $url['path'] ) {
+					return new Response(
+						200,
+						array(),
+						json_encode(
+							$property
+						)
+					);
+				}
+
+				return new Response( 200 );
+			}
+		);
+	}
+
+	public function test_register() {
+		remove_all_actions( Synchronize_Property::CRON_SYNCHRONIZE_PROPERTY );
+
+		$this->synchronize_property->register();
+		$this->synchronize_property->maybe_schedule_synchronize_property();
+
+		$this->assertTrue( has_action( Synchronize_Property::CRON_SYNCHRONIZE_PROPERTY ) );
+
+		$this->assertTrue(
+			(bool) wp_next_scheduled( Synchronize_Property::CRON_SYNCHRONIZE_PROPERTY )
+		);
+	}
+
+	public function test_schedule_synchronize_property__cron_callback() {
+		$property_id    = '987654321';
+		$create_time    = '2014-10-02T15:01:23Z';
+		$create_time_ms = Synchronize_Property::convert_time_to_unix_ms( $create_time );
+
+		$this->authentication->get_oauth_client()->set_granted_scopes(
+			$this->analytics_4->get_scopes()
+		);
+
+		$this->fake_analytics_4_property_object( $property_id, $create_time );
+
+		// Confirm that user is authenticated and module is connected, as it will be needed in the cron callback.
+		$this->assertTrue( $this->authentication->is_authenticated() );
```

Comment:
This is what I had mentioned in the last review which is in the context of a normal cron run, there is no logged in user, so we wouldn't want to set one beforehand since that would give us a false positive if we hadn't set the user properly in our sync method. Our cron function requires that the module owner is authenticated but we can only really infer this from the success of the action. If we really wanted to (probably overkill here) we could check the request that is made to see that it is using the access token of the module owner, but I don't think it's necessary.